src/main/java/com/aws/greengrass/mqtt/bridge/MessageBridge.java

```diff
@@ -50,21 +51,38 @@ public class MessageBridge {
      */
     public MessageBridge(TopicMapping topicMapping) {
         this.topicMapping = topicMapping;
-        this.topicMapping.listenToUpdates(this::processMappingAndSubscribe);
+    }
+
+    /**
+     * Start message bridge.
+     *
+     * <br><br><p>All registered {@link MessageClient}s will have their
+     * subscriptions updated immediately.  Changes to {@link TopicMapping} will
+     * trigger subscription updates as well.
+     */
+    public void start() {
+        topicMapping.listenToUpdates(updateListener);
         processMappingAndSubscribe();
     }
 
+    /**
+     * Stop the message bridge.
+     */
+    public void stop() {
+        topicMapping.unsubscribeFromUpdates(updateListener);
```

Comment:
not clear to me that we do want to unsubscribe here. I feel that there's a potential for missing changes if we do this. As long as you only subscribe once or even if you subscribe multiple times but with the same listener, there won't be multiple subscriptions.